### PR TITLE
chore!: upgrade to penumbra release 0.66

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Updates to the newest penumbra release which is being used for their testnet.

## Background
Penumbra recently upgraded their testnet, and intends to support network upgrades on their new network. This updates to that version, which should mean we can provide network upgrades with future penumbra updates hopefully.

## Changes
- Update to latest penumbra release.
- updated readme around a direction that didn't work for me.

## Testing
Automated tests, manually started up sequencer locally to ensure starts.

## Breaking Changelist
- Upgrades to cnidarium & penumbra are generally breaking changes
- Added IBC params to genesis

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -66,11 +66,11 @@ once_cell = "1.17.1"
 sha2 = "0.10"
 serde = "1"
 serde_json = "1"
-# bump to a release after https://github.com/penumbra-zone/penumbra/pull/3598 is included
+# Corresponds with penumbra release 0.66
 # Note that when updating the penumbra versions, vendored types in `proto/sequencerapis/astria_vendored` may need to be updated as well.
-penumbra-ibc = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "1b1be7e1373401eaf30d63352ee680991cb42aea", default-features = false }
-penumbra-proto = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "1b1be7e1373401eaf30d63352ee680991cb42aea" }
-penumbra-tower-trace = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "1b1be7e1373401eaf30d63352ee680991cb42aea" }
+penumbra-ibc = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "29b93231c6dc4a534b7a1177790d3d39d2dfb52c", default-features = false }
```
Comment (SuperFluffy) [inline Cargo.toml:71]: Prefer the `tag` key. We used `rev` because we needed a not-yet (tag-)release feature:
```suggestion
penumbra-ibc = { git = "https://github.com/penumbra-zone/penumbra.git", tag = "v0.66.0", default-features = false }
```

And same for the others below

File: Cargo.toml [hunk 2]
```diff
@@ -66,11 +66,11 @@ once_cell = "1.17.1"
 sha2 = "0.10"
 serde = "1"
 serde_json = "1"
-# bump to a release after https://github.com/penumbra-zone/penumbra/pull/3598 is included
+# Corresponds with penumbra release 0.66
```
Comment (SuperFluffy) [inline Cargo.toml:69]: Remove, see comment below

File: crates/astria-sequencer/Cargo.toml [hunk 3]
```diff
@@ -21,8 +21,9 @@ matchit = "0.7.2"
 tower = "0.4"
 tower-abci = "0.11.0"
 tower-actor = "0.1.0"
-cnidarium = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "1b1be7e1373401eaf30d63352ee680991cb42aea" }
-cnidarium-component = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "1b1be7e1373401eaf30d63352ee680991cb42aea" }
+# These correspond to penumbra 0.66 release
+cnidarium = { git = "https://github.com/penumbra-zone/penumbra.git", rev = "29b93231c6dc4a534b7a1177790d3d39d2dfb52c" }
```
Comment (SuperFluffy) [inline crates/astria-sequencer/Cargo.toml:25]: As for the penumbra depdencies in the workspace: prefer `tag` here and on the line below.

File: crates/astria-sequencer/README.md [hunk 4]
```diff
@@ -105,7 +105,8 @@ cometbft init
 # application state
 ../../target/debug/astria-sequencer-utils \
     --genesis-app-state-file=test-genesis-app-state.json \
-    --destination-genesis-file=$HOME/.cometbft/config/genesis.json
+    --destination-genesis-file=$HOME/.cometbft/config/genesis.json \
+    --chain-id <desired_test_chain_id>
```
Comment (SuperFluffy) [inline crates/astria-sequencer/README.md:109]: Very nit: seems orthogonal to the bump but makes sense to do this here (given that your testing didn't work).

File: crates/astria-sequencer/src/genesis.rs [hunk 5]
```diff
@@ -13,6 +14,7 @@ pub(crate) struct GenesisState {
     #[serde(deserialize_with = "deserialize_address")]
     pub(crate) ibc_sudo_address: Address,
     pub(crate) native_asset_base_denomination: String,
+    pub(crate) ibc_params: IBCParameters,
```
Comment (SuperFluffy) [inline crates/astria-sequencer/src/genesis.rs:17]: Please mention this in the PR message that the genesis file was changed.

**Review by SuperFluffy — APPROVED** [review]
Just some small requests around documentation and style of the dependency declaration.

File: crates/astria-sequencer/src/genesis.rs (hunk 5 above)
Comment (joroshiba) [reply crates/astria-sequencer/src/genesis.rs:17]: Already there?

Comment (SuperFluffy) [reply crates/astria-sequencer/src/genesis.rs:17]: Yeah, there at the bottom in breaking changes. Didn't see 🙏
